### PR TITLE
Prohibit importing built_value using 'show' or 'as'.

### PR DESCRIPTION
**Pull request by davidmorgan**
This change is [Reviewable](https://reviewable.io/reviews/google/built_value.dart/332)

### CODE REVIEW

File: built_value_generator/lib/src/value_source_class.dart [hunk 1]
```diff
@@ -99,7 +108,25 @@ abstract class ValueSourceClass
...
+    return source.contains(expectedCode);
+  }
+
+  @memoized
+  bool get hasBuiltValueImportWithShow {
+    // It would be more accurate to check using the AST, but this is
+    // potentially expensive. We already have the source for the "part of"
+    // check, use that.
+    return source.contains('$_importWithSingleQuotes show') ||
+        source.contains('$_importWithSingleQuotes show');
```
Comment (manuelF) [inline built_value_generator/lib/src/value_source_class.dart]: _importWithDoubleQuotes

File: built_value_generator/lib/src/value_source_class.dart [hunk 2]
```diff
@@ -203,6 +230,16 @@ abstract class ValueSourceClass
       result.add('Make class abstract.');
     }
 
+    if (hasBuiltValueImportWithShow) {
+      result.add('Stop using "show" when importing '
+          '"package:built_value/built_value.dart".');
```
Comment (manuelF) [inline built_value_generator/lib/src/value_source_class.dart]: Maybe add a reason for it?

File: built_value_generator/lib/src/value_source_class.dart (hunk 1 above)
Comment (davidmorgan) [reply built_value_generator/lib/src/value_source_class.dart]: Thanks! I already found and fixed that on a different revision at least once. Now added a test for it like I should have done in the first place...

File: built_value_generator/lib/src/value_source_class.dart (hunk 2 above)
Comment (davidmorgan) [reply built_value_generator/lib/src/value_source_class.dart]: Done: 'it prevents the generated code from finding helper methods'

**Pull request merged**